### ISSUE TITLE
uk_test: self-tests fail on kvm / arm64

### ISSUE THREAD

**Issue by michpappas** [unikraft/unikraft#393]
### Describe the bug

self-tests fail on kvm / arm64 due to an assertion. The assertion is triggered in `_uk_test_do_assert()`, and is caused by  `_uk_test_find_assert()` failing to find an assertion by line number.

### Steps to reproduce

Can reproduce with app-helloworld. Run menuconfig and apply the following changes:
* arch: Armv8
* plat: KVM
* uk_test: enable
 
Using gcc-arm-10.3-2021.07-x86_64-aarch64-none-elf.

### Which platforms were you using or does this bug affect?

kvm

### Relevant log output

```shell
app-helloworld$ qemu-system-aarch64 -machine virt -cpu max -monitor none -serial stdio -kernel build/app-helloworld_kvm-arm64
test: uktest_myself_testsuite->uktest_test_sanity
    :	expected `((void *) 0)` to be 0 and was 0 ....................................... [ PASSED ]
[    0.000000] CRIT: [libuktest] <myself.c @  653> Assertion failure: assert
```

**Comment by marcrittinghaus** [comment]
Hi @michpappas,

I can reproduce the issue. The assert is raised as the test code cannot find the test assertion in the assert table. It searches for the line number in the table. The table is an array of `struct uk_assert`, which is defined as:
```C
struct uk_assert {
	/* The status of the assertion (one of UK_TEST_ASSERT_*) */
	unsigned short status;
	/* The line of the assertion in the source code file */
	unsigned short line;
} __packed;
```
The table entry for the first self test in `myself.c` is created with this snippet (expanded):
```ASM
.pushsection ".uk_asserttab", 45, "a"
.word 0
.word 45
.popsection
```
Interestingly, the ARM compiler generates the following:
```
(gdb) p *assert
$1 = {status = 45, line = 0}
```
I am not sure why it reorders the fields 🤔 I will look for a solution...

Seems `uktest` has not been properly tested 😆 Sorry for the inconvenience 🙏🏻

**Comment by nderjung** [comment]
Fixed in #397
